src/main/java/domain/piece/state/ContinuousPiece.java

```diff
@@ -0,0 +1,63 @@
+package domain.piece.state;
+
+import static domain.pattern.Direction.DOWN;
+import static domain.pattern.Direction.LEFT;
+import static domain.pattern.Direction.RIGHT;
+import static domain.pattern.Direction.UP;
+
+import domain.JanggiPosition;
+import domain.pattern.Direction;
+import domain.pattern.Path;
+import domain.pattern.Pattern;
+import java.util.Collections;
+import java.util.List;
+
+public abstract class ContinuousPiece extends Moved {
+    public List<Pattern> findMovablePath(Path path, JanggiPosition beforePosition, JanggiPosition afterPosition) {
+        if (afterPosition.file() == beforePosition.file()) {

```

Comment:
```suggestion
        if (beforePosition.isSameFileWith(afterPosition)) {
```
해당 객체가 직접 비교를 책임지도록 해볼까요?